UnityProject/Assets/Scripts/ScriptableObjects/GhostRoleList.cs

```diff
@@ -1,19 +1,37 @@
 ﻿using System.Collections.Generic;
 using UnityEngine;
 using NaughtyAttributes;
+using Antagonists;
+using Logs;
 
 namespace ScriptableObjects
 {
 	/// <summary>
 	/// Contains a list of all possible ghost roles. This information is available on both server and client.
 	/// </summary>
 	[CreateAssetMenu(fileName = "GhostRoleList", menuName = "ScriptableObjects/Systems/GhostRoles/GhostRoleList")]
-	public class GhostRoleList : ScriptableObject
+	public class GhostRoleList: SingletonScriptableObject<GhostRoleList>
 	{
 		[Tooltip("A list of roles that are potentially available to ghosts during a round.")]
 		[SerializeField, ReorderableList]
 		private List<GhostRoleData> ghostRoles = new List<GhostRoleData>();
 
 		public List<GhostRoleData> GhostRoles => ghostRoles;
+
+		public short GetIndex(GhostRoleData ghostRole)
```

Comment:
I think it would be better to have a unique ID instead of handling them by index